DataStructure/src/main/list/ResizableArray.java

```diff
@@ -135,7 +161,7 @@ public boolean contains(int num){
      * @return the index of the last occurrence of the given element, -1 otherwise.
      */
     public int lastIndexOf(int num){
-        for (int i = writeIndex; i >= 0; i--){
+        for (int i = writeIndex - 1; i >= 0; i--){
             if (num == list[i]){
                 return i ;
             }
```

Comment:
The java doc of clear method is incorrect - "Removes all the elements but keeps the size". Haha, How can you keep the size after removing all the elements ? = Insincerity